src/capabilities.js

```diff
@@ -280,11 +281,23 @@ export class Capabilities {
     if (!ownCapabilities.roleAssignment.includes(roleId)) {
       throw new Error('No capability to assign role ' + roleId)
     }
-    await this.#dataType[kCreateWithDocId](deviceId, {
-      schemaName: 'role',
-      roleId,
-      fromIndex,
-    })
+
+    const existingRoleDoc = await this.#dataType
+      .getByDocId(deviceId)
+      .catch(() => null)
+
+    if (existingRoleDoc) {
+      await this.#dataType.update(existingRoleDoc.versionId, {
```

Comment:
Wondering if this should be a merge update? E.g.

```js
const links = [existingRoleDoc.versionId, ...existingRoleDoc.forks]
await this.#dataType.update(links, //...
```

I think maybe yes for now, to keep the database clean, until we have some kind of UI to deal with forks in a role.

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/423/commits/120eea5c700ca5b1b1244e1a6ac316ec72b19aac